File: minesweeper/python/test_timer.py
from time import sleep
import unittest
from Timer import Timer

class TestTimer(unittest.TestCase):

    def test_timer_can_start(self):
        timer = Timer()
        timer.start()

    def test_timer_can_query_seconds_elapsed_after_starting(self):
        timer = Timer()
        timer.start()
        sleep(1)
        self.assertEqual(timer.seconds_elapsed(), 1)
        sleep(2)
        self.assertEqual(timer.seconds_elapsed(), 3)

File: minesweeper/dart/test_timer.dart
import 'package:test/test.dart';
import 'dart:io';
import './Timer.dart';

void main() {
  test('Timer can start', () {
    Timer timer = Timer();
    timer.start();
  });

  test('Timer can query seconds elapsed after starting', () {
    Timer timer = Timer();
    timer.start();
    sleep(const Duration(seconds:1));
    expect(timer.seconds_elapsed(), 1);
    sleep(const Duration(seconds:2));
    expect(timer.seconds_elapsed(), 3);
  });
}
